charts/corda/templates/bootstrap.yaml

```
@@ -2,3 +2,13 @@
 {{- include "corda.bootstrapDbJob" . }}
 {{- include "corda.bootstrapKafkaJob" . }}
 {{- include "corda.bootstrapRbacJob" . }}
+{{- if .Values.bootstrap.db.enabled }}
+{{-   range $workerName, $authConfig := .Values.bootstrap.db.stateManager }}
+{{-     $workerConfig := (index $.Values.workers $workerName) }}
+{{-     if and (not $workerConfig.stateManager.db.host) (ne $authConfig.username.value "") (not $authConfig.valueFrom) -}}
```

`""` is falsy and I think you need to look all the way down to the secret name making the logic something like:
```suggestion
{{-     if and ( not $workerConfig.stateManager.db.host ) ( not $authConfig.username.value ) ( not $authConfig.valueFrom.secretKeyRef.name ) -}}
```

---

Done (used another approach and improved the condition, though)